dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -749,6 +752,379 @@ def test_min_max_id_properly_update(self, *mocks):
+        profiler_options.set({'data_labeler.is_enabled': False})
+
+        cls.profiler = UnstructuredProfiler(
+            cls.dataset, len(cls.dataset), options=profiler_options)
+        cls.profiler2 = UnstructuredProfiler(
+            pd.DataFrame(['extra', '\n', 'test\n', 'data .', 'For merging.']),
+            options=profiler_options
+        )
+        cls.report = cls.profiler.report()
+
+    def test_sample(self):
+        self.maxDiff = None
+        self.assertCountEqual(
+            ['Report',
+             'Reply',
+             'Removing duplication that repeats the handling of the exact same '
+             'business rule is also usually a win.',
+             'edited 9 hours ago',
+             'Just want to caution against following this too rigidly.'],
+            self.profiler.sample
+        )
+
+    def test_total_samples(self):
+        self.assertEqual(16, self.profiler.total_samples)
+
+    def test_empty_line_count(self):
+        self.assertEqual(7, self.profiler._empty_line_count)
+
+    def test_text_profiler_results(self):
+        # pop out times
+        self.assertIsNotNone(
+            self.report['data_stats']['statistics'].pop('times'))
+
+        # vocab order doesn't matter
+        expected_vocab = ['x', 'i', 'y', 's', '9', ',', 'u', 'b', 'f', 'Y', 'J',
+                          'v', 'r', 'o', 'a', '6', 'n', 'h', ' ', 'g', 'R', 't',
+                          'W', '.', 'm', 'c', 'l', 'e', 'p', 'w', 'S', 'd', 'D',
+                          ':']
+        self.assertCountEqual(
+            expected_vocab,
+            self.report['data_stats']['statistics'].pop('vocab'))
+
+        # vocab order doesn't matter, case insensitive, remove stop words
+        # assuming if words are correct, rest of TextProfiler is merged properly
+        # vocab order doesn't matter, case insensitive, remove stop words
+        expected_word_count = {
+            'edited': 1, '9': 1, 'hours': 1, 'ago': 1, '6': 1, 'Do': 1,
+            'not': 1, 'duplicate': 1, 'code': 2, 'Just': 1, 'want': 1,
+            'to': 2, 'caution': 1, 'against': 1, 'following': 1, 'this': 1,
+            'too': 1, 'rigidly': 1, 'When': 1, 'you': 2, 'try': 1, 'DRY': 1,
+            'them': 1, 'up': 1, 'into': 1, 'a': 2, 'single': 1, 'generic': 1,
+            'abstraction': 1, 'have': 1, 'inadvertently': 1, 'coupled': 1,
+            'those': 1, 'two': 1, 'business': 3, 'rules': 1, 'together': 1,
+            'Removing': 1, 'duplication': 1, 'that': 1, 'repeats': 1, 'the': 2,
+            'handling': 1, 'of': 1, 'exact': 1, 'same': 1, 'rule': 1, 'is': 1,
+            'also': 1, 'usually': 1, 'win': 2, 'Duplicate': 1, 'words': 1,
+            'Reply': 1, 'Share': 1, 'Report': 1}
+
+        # adapt to the stop words (brittle test)
+        stop_words = \
+            self.profiler._profile._profiles['unstructured_text']._stop_words
+        for key in list(expected_word_count.keys()):
+            if key.lower() in stop_words:
```

Comment:
why we need to lower the key here? When we compare with stop_words we should keep the words as is?

Reply:
stop words are lower, hence matching the key which is case sensitive needs matching

Reply:
we don't actually alter the key in the dict